torch/utils/_pytree.py

```diff
@@ -951,11 +951,16 @@ def unflatten(self, leaves: Iterable[Any]) -> PyTree:
         return unflatten_fn(child_pytrees, self.context)
 
 
+@dataclasses.dataclass(frozen=True, repr=False)
 class LeafSpec(TreeSpec):
-    def __init__(self) -> None:
-        super().__init__(None, None, [])
+    type: Any = dataclasses.field(default=None, init=False)
```

Comment:
This should be Optional[Any] or Optional[Type[Any]], right?

Reply:
I think it's a historical conservative typing from `TreeSpec`: https://github.com/pytorch/pytorch/blob/89532720f63de371a91e385611c23e93dbe82c6f/torch/utils/_pytree.py#L808-L810